exporters/prometheus/exporter.go

```diff
@@ -386,16 +387,63 @@ func sanitizeName(n string) string {
 	return b.String()
 }
 
-func validateMetrics(name, description string, metricType *dto.MetricType, mfs map[string]*dto.MetricFamily) (drop bool, help string) {
-	emf, exist := mfs[name]
+func (c *collector) metricTypeAndName(m metricdata.Metrics) (*dto.MetricType, string) {
+	name := c.getName(m)
+
+	switch v := m.Data.(type) {
+	case metricdata.Histogram[int64], metricdata.Histogram[float64]:
+		return dto.MetricType_HISTOGRAM.Enum(), name
+	case metricdata.Sum[float64]:
+		if v.IsMonotonic {
+			return dto.MetricType_COUNTER.Enum(), name + counterSuffix
+		}
+		return dto.MetricType_GAUGE.Enum(), name
+	case metricdata.Sum[int64]:
```

Comment:
can this case be combined with the other Sum case?

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/4270

Reply:
@dashpole Do you mean e.g.:

```go
case metricdata.Histogram[int64], metricdata.Histogram[float64]:
				addHistogramMetric(ch, v, m, keys, values, name)
```

?

If so then the compiler cannot handle this case: 

> type metricdata.Aggregation of v does not match metricdata.Histogram[N] (cannot infer N)compiler[CannotInferTypeArgs](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#CannotInferTypeArgs)

Reply:
Thanks